yt_dlp/extractor/facebook.py

```diff
@@ -478,9 +480,35 @@ def _perform_login(self, username, password):
             return
 
     def _extract_from_url(self, url, video_id):
+        cookies = self._get_cookies(url)
+        # user passed logged-in cookies or attempted to login
+        login_data = cookies.get('c_user') and cookies.get('xs')
+        logged_in = False
+
         webpage = self._download_webpage(
             url.replace('://m.facebook.com/', '://www.facebook.com/'), video_id)
 
+        sjs_data = [self._parse_json(j, video_id, fatal=False) for j in re.findall(
+            r'data-sjs>({.*?ScheduledServerJS.*?})</script>', webpage)]
+
+        if login_data:
+            logged_in = get_first(sjs_data, (
+                'require', ..., ..., ..., '__bbox', 'define',
+                lambda _, v: 'CurrentUserInitialData' in v, ..., 'ACCOUNT_ID'), default='0') != '0'
+            if logged_in:
+                if any(content in webpage for content in ['180 days left to appeal', 'suspended your account']):
+                    raise ExtractorError('Your account is suspended', expected=True)
+                if 'send a code to confirm the mobile number you give us' in webpage:
+                    raise ExtractorError('Facebook is requiring mobile number confirmation', expected=True)
+                if 'your account has been locked' in webpage:
```

Comment:
These checks are problematic since they could be phrases found in user-posted content. Not sure how necessary this is either

Reply:
Changed those to be based quite specifically on _js_ data.